torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -356,6 +356,21 @@ TCPStore::TCPStore(std::string host, const TCPStoreOptions& opts)
     addr_.port = opts.port;
   }
 
+  if (opts.connectAsync) {
+    connectFuture_ =
+        std::async(std::launch::async, [this, opts] { connect(opts); });
+  } else {
+    connect(opts);
+  }
+
+  if (opts.waitWorkers) {
```

Comment:
What do you think making a single control via waitForworkers -> true: sync, wait for workers. false: async?

Reply:
I think that's reasonable -- will need to add an env killswitch to the async behavior though

I'm a bit nervous about rolling this out